extensions/flight-sql/src/main/java/io/deephaven/server/flightsql/FlightSqlResolver.java

```diff
@@ -0,0 +1,1710 @@
+     */
+    @Override
+    public <T> SessionState.ExportBuilder<T> publish(
+            final SessionState session,
+            final ByteBuffer ticket,
+            final String logId,
+            @Nullable final Runnable onPublish) {
+        if (session == null) {
+            throw unauthenticatedError();
+        }
+        throw error(Code.FAILED_PRECONDITION,
+                "Could not publish '" + logId + "': Flight SQL tickets cannot be published to");
+    }
+
+    // ---------------------------------------------------------------------------------------------------------------
+
+    @Override
+    public String getLogNameFor(final ByteBuffer ticket, final String logId) {
+        // This is a bit different from the other resolvers; a ticket may be a very long byte string here since it
+        // may represent a command.
+        return FlightSqlTicketHelper.toReadableString(ticket, logId);
+    }
+
+    // ---------------------------------------------------------------------------------------------------------------
+
+    interface CommandHandler<C> {
+
+        TicketHandler initialize(C command);
+    }
+
+    interface TicketHandler {
+
+        boolean isOwner(SessionState session);
+
+        FlightInfo getInfo(FlightDescriptor descriptor);
+
+        Table resolve();
+    }
+
+    interface TicketHandlerReleasable extends TicketHandler {
+
+        void release();
+    }
+
+    private Table executeSqlQuery(SessionState session, String sql) {
+        // See SQLTODO(catalog-reader-implementation)
+        final QueryScope queryScope = ExecutionContext.getContext().getQueryScope();
+        // noinspection unchecked,rawtypes
+        final Map<String, Table> queryScopeTables =
+                (Map<String, Table>) (Map) queryScope.toMap(queryScope::unwrapObject, (n, t) -> t instanceof Table);
+        final TableSpec tableSpec = Sql.parseSql(sql, queryScopeTables, TicketTable::fromQueryScopeField, null);
+        // Note: this is doing io.deephaven.server.session.TicketResolver.Authorization.transform, but not
+        // io.deephaven.auth.ServiceAuthWiring
+        // TODO(deephaven-core#6307): Declarative server-side table execution logic that preserves authorization logic
+        try (final SafeCloseable ignored = LivenessScopeStack.open()) {
+            final Table table = tableSpec.logic()
+                    .create(new TableCreatorScopeTickets(TableCreatorImpl.INSTANCE, scopeTicketResolver, session));
+            if (table.isRefreshing()) {
+                table.retainReference();
+            }
+            return table;
+        }
+    }
+
+    /**
+     * This is the base class for "easy" commands; that is, commands that have a fixed schema and are cheap to
+     * initialize.
+     */
+    static abstract class CommandHandlerFixedBase<T extends Message> implements CommandHandler<T> {
+        private final Class<T> clazz;
```

Comment:
field is unused - but we require it to be non-null, and require subtypes to pass a type that ostensibly matches it?

Reply:
Removed.